caddyhttp/httpserver/replacer.go

```diff
@@ -358,7 +358,12 @@ func (r *replacer) getSubstitution(key string) string {
 		if r.responseRecorder == nil {
 			return r.emptyValue
 		}
-		return roundDuration(time.Since(r.responseRecorder.start)).String()
+		elapsedDuration := time.Since(r.responseRecorder.start)
+
+		if elapsedDuration == 0 {
+			return "0ms"
+		}
```

Comment:
time.Since returns a duration value, which has accuracy to the nanosecond, I believe. So while 0ms is technically true in that case, the likelihood of getting an actual value of 0 is extremely low, I think. Is there a test that exhibits this behavior?

Reply:
I get 0s durations for about 40% of status 404 and 20% of status 200 on pages with little content.

I'm unsure if this issue affects only Windows 7 x64 or if it affects other Operating Systems. I'm unable to test on others at the moment.

I'm unsure if a test that is valid 100% of the time would be doable. I think it may need more user testing first.

Reply:
I do wonder if that's OS-dependent. time.Since should return granularity down to the nanosecond, at least, as far as Go is concerned: we'll have to try to find out if your OS is what's causing a value of 0 to be returned.